src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -145,6 +151,11 @@ private void validateSchema(final JSONObject event, final EventType eventType) t
         }
     }
 
+    private void validateEventSize(final BatchItem item) throws EventValidationException {
+        if (item.getEvent().toString().getBytes().length > nakadiSettings.getEventMaxBytes())
```

Comment:
Another problem is to actually to do that conversion. There is the byte array representation that you can use. Or you should store the size into the item to avoid that conversion.